src/launcher.rs

```diff
@@ -396,11 +406,42 @@ impl Launcher {
         Ok(())
     }
 
+    /// Runs the shutdown script if it exists and exits the node-launcher process
+    /// with the exit code returned by the script, otherwise returns 0.
+    fn run_shutdown_script_and_exit(&mut self) -> Result<()> {
+        let exit_code = if Path::new(SHUTDOWN_SCRIPT_PATH).exists() {
+            info!("Running shutdown script at {}.", SHUTDOWN_SCRIPT_PATH);
+            let status = utils::map_and_log_error(
+                Command::new(SHUTDOWN_SCRIPT_PATH).status(),
+                format!("Couldn't execute script at {}", SHUTDOWN_SCRIPT_PATH),
+            )?;
+            status.code().unwrap_or_else(|| {
+                error!("Shutdown script was terminated by a signal.");
+                1
```

Comment:
This is tricky... The user that examines the return code of the launcher won't be able to tell whether `1` originated from the script or from the launcher.

It's worth checking with business (or with whoever requested the ticket) if we can reasons anything about the possible values returned by the script.

Anyway, I think we should maybe go for some less common value than `1` here, maybe, for example `254`? Rationale: `1` or other values close to 0 would probably be chosen by the script author to indicate error conditions.

Reply:
I agree with this. As per the other comment, we need some convention here to establish that a signal killed the process. `1` is not the best choice here. I would have hoped someone would have suggestions during review :grin: . I'll check in with the ticket requester.

Reply:
Changed to `254` for now, let's see what other reviewers think.